### PR TITLE
Provision latch

### PR DESCRIPTION
**Pull request by sjcorbett**
First commit is boring - it just simplifies `JcloudsLocationTest`.

Second commit adds `BrooklynConfigKeys.PROVISION_LATCH`, to block provisioning of locations until some criteria is met.

### CODE REVIEW

**Comment by sjcorbett** [comment]
cc @alasdairhodge

File: core/src/main/java/brooklyn/entity/basic/BrooklynConfigKeys.java [hunk 1]
```diff
@@ -143,6 +143,7 @@
      * component is up, but this entity does not care about the dependent component's actual config values.
      */
 
+    public static final ConfigKey<Boolean> PROVISION_LATCH = newBooleanConfigKey("provision.latch", "Latch for blocking location provision until ready");
```
Comment (alasdairhodge) [inline core/src/main/java/brooklyn/entity/basic/BrooklynConfigKeys.java]: `SoftwareProcess` redefines the other lifecycle latches (referencing the ones here in `BrooklynConfigKeys`); should `PROVISION_LATCH` also be aliased there for easier discovery? I'm not convinced this is necessarily a good pattern, but inconsistency is worse.

Comment (alasdairhodge) [inline core/src/main/java/brooklyn/entity/basic/BrooklynConfigKeys.java]: Oh wait, `SoftwareProcess` itself isn't consistent with which keys it aliases. Ignore me.

Comment (sjcorbett) [inline core/src/main/java/brooklyn/entity/basic/BrooklynConfigKeys.java]: I thought about it but figured the provisioning latch doesn't relate to the software-process-ness of an entity.

Comment (alasdairhodge) [inline core/src/main/java/brooklyn/entity/basic/BrooklynConfigKeys.java]: Yeah, I agree. I expect `PROVISION_LATCH` will eventually join `PROVISIONING_PROPERTIES` on some sort of `Provisionable` trait class.

**Comment by alasdairhodge** [comment]
Nice addition. LGTM.

**Pull request merged**